src/Utilities/ErrorHandling/FormatStacktrace.hpp

```diff
@@ -0,0 +1,87 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Formatter for Boost stacktrace, following these docs:
+/// https://www.boost.org/doc/libs/1_78_0/doc/html/stacktrace/getting_started.html#stacktrace.getting_started.global_control_over_stacktrace_o
+
+#pragma once
+
+#include <boost/stacktrace/stacktrace_fwd.hpp>
+
+#include <cstddef>
+#include <ostream>
+#include <string>
+
+namespace stacktrace_detail {
+std::string abbreviated_symbol_name(const std::string& symbol_name);
+std::string get_stack_frame(void* addr);
+#if __has_include(<link.h>)
+std::string addr2line_command(const void* addr);
+#endif
+}  // namespace stacktrace_detail
+
+namespace boost {
+namespace stacktrace {
+
+template <class CharT, class TraitsT, class Allocator>
+std::basic_ostream<CharT, TraitsT>& do_stream_st(
+    std::basic_ostream<CharT, TraitsT>& os,
+    const basic_stacktrace<Allocator>& backtrace) {
+  const std::streamsize standard_width = os.width();
+  const size_t frames = backtrace.size();
+  for (size_t i = 0; i < frames; ++i) {
+    const auto& frame = backtrace[i];
+    const std::string& symbol_name = frame.name();
+    const std::string& source_file = frame.source_file();
+    const size_t source_line = frame.source_line();
+    // Enumerate frame number
+    os.width(3);
+    os << i;
+    os.width(standard_width);
+    os << ". ";
+    if (frame.empty()) {
+      os << "[empty]";
+      continue;
+    }
+    // Skip frames originating in error handling code
+    if (symbol_name.find("abort_with_error_message") != std::string::npos or
+        source_file.find("src/Utilities/ErrorHandling/") != std::string::npos or
+        symbol_name.find("boost::stacktrace") != std::string::npos) {
+      os << "[error handling]\n";
+      continue;
+    }
+    if (symbol_name.empty()) {
+      // Boost was unable to get the symbol name (probably because dladdr can't
+      // find it either). Fall back to printing the default stack frame.
+      os << stacktrace_detail::get_stack_frame(
+          const_cast<void*>(frame.address()));
+    } else {
+      // Print symbol name. Abbreviate if necessary to avoid filling the screen
+      // with templates.
+      os << stacktrace_detail::abbreviated_symbol_name(symbol_name);
+    }
+    if (source_line != 0) {
+      // Print location in source file if available
+      os << " in " << source_file << ":" << source_line;
+    } else {
+#if __has_include(<link.h>)
+      // Print addr2line information otherwise
+      os << " - Resolve with: "
+         << stacktrace_detail::addr2line_command(frame.address());
+#endif
```

Comment:
Should we print a note that we couldn't get the line? I don't know if that's a good idea since it could clutter things. Though maybe at the very end of the trace we print "Couldn't include line number info". It's sort of obvious by them missing, but it's hard to know you're missing something if you didn't know it should be there. I'm not sure, just something to think about :shrug:

Reply:
I edited this output slightly to read "Resolve source file and line with:". That's not much longer but gives some more info.